packages/client/src/createTRPCClient.ts

```diff
@@ -78,12 +78,14 @@ type DecorateProcedure<
  */
 type DecoratedProcedureRecord<
   TRouter extends AnyRouter,
-  TProcedures extends ProcedureRouterRecord,
+  TRecord extends RouterRecord,
 > = {
-  [TKey in keyof TProcedures]: TProcedures[TKey] extends AnyRouter
-    ? DecoratedProcedureRecord<TRouter, TProcedures[TKey]['_def']['record']>
-    : TProcedures[TKey] extends AnyProcedure
-    ? DecorateProcedure<TRouter['_def']['_config']['$types'], TProcedures[TKey]>
+  [TKey in keyof TRecord]: TRecord[TKey] extends infer $Value
```

Comment:
why do you need this `extends infer` here?

Reply:
I just think it makes the "looping" clearer

Reply:
Only need to grab `TRecord[Key]` once, it's like a declaring a variable

Reply:
> Only need to grab `TRecord[Key]` once, it's like a declaring a variable

aah i see, how is the compiler performance comparatively?

Reply:
No idea tbh 🙈